dsgrid/config/dimension_associations.py

```diff
@@ -0,0 +1,165 @@
+import itertools
+import logging
+from pathlib import Path
+from typing import Dict
+
+from dsgrid.dimension.base_models import DimensionType
+from dsgrid.exceptions import DSGInvalidDimensionAssociation
+from dsgrid.utils.spark import read_dataframe
+
+
+logger = logging.getLogger(__name__)
+
+
+class DimensionAssociations:
+    """Interface to a project's dimension associations"""
+
+    def __init__(self, associations: Dict):
+        # This has the format {(DimensionType1, DimensionType2, ..., DimensionTypeN): df}
+        # The keys are sorted by DimensionType so that the caller's order doesn't matter.
+        self._associations = associations
+
+    def __bool__(self):
+        return bool(self._associations)
+
+    def __len__(self):
+        return len(self._associations)
+
+    @classmethod
+    def load(cls, path: Path, association_files):
+        """Load dimension associations from a path.
+
+        Parameters
+        ----------
+        path: Path
+        association_files : list
+            List of filenames with paths relative to path
+
+        Returns
+        -------
+        DimensionAssociations
+
+        """
+        associations = {}
+        for association_file in association_files:
+            filename = path / association_file
+            records = read_dataframe(filename, cache=True)
+            types = tuple(DimensionType(x) for x in sorted(records.columns))
+            associations[types] = records
+            logger.debug("Loaded dimension associations from %s %s", path, records.columns)
+
+        return cls(associations)
+
+    def get_associations(self, *dimensions):
+        """Return the records for this dimension association.
+
+        Parameters
+        ----------
+        dimensions : tuple
+            Any number of instances of DimensionType
```

Comment:
This file does not use `dimension_type`. I used `dimension` here for brevity because there shouldn't be any confusion with `dimension_type` vs `dimension_config`, which could exist in other files. I still prefer the shorter name but am open to changing it.